packages/dds/map/src/directory.ts

```diff
@@ -467,16 +467,16 @@ export class SharedDirectory
 	 * @param id - Optional name of the shared directory
 	 * @returns Newly create shared directory (but not attached yet)
 	 */
-	public static create(runtime: IFluidDataStoreRuntime, id?: string): SharedDirectory {
-		return runtime.createChannel(id, DirectoryFactory.Type) as SharedDirectory;
+	public static create(runtime: IFluidDataStoreRuntime, id?: string): ISharedDirectory {
+		return runtime.createChannel(id, DirectoryFactory.Type) as ISharedDirectory;
```

Comment:
I know this may be out of scope, I'm liking this PR already. I wonder if we could add types to `ISharedDirectory<T>` and `ISharedMap<T>`

That would be awesome :).

Reply:
There are complexities with doing that since that T might want to be changed over time, which brings in compatibility issues. SharedTree is building quite a lot of stuff to properly support something like that, but I think maybe applying the actual type parameter belongs in a data object not the DDS for some nuanced reasons around enabling flexibility for handling compatibility errors. Also such typing really should be parried with persisting type information in the document to validate you don't get compatibility errors on accident: SharedTree also does this.

So in short, yes we should aim for that, but its persisted data so its complicated and needs a good compatibility and schema migration story. I'll make sure tree delivers that in a way that works cleanly with container schema, then maybe we can cover the other DDSes.